cmd/iaviewer/main.go

```diff
@@ -22,41 +21,68 @@ const (
 	DefaultCacheSize int = 10000
 )
 
+var cmds = map[string]bool{
+	"data":      true,
+	"data-full": true,
+	"hash":      true,
+	"shape":     true,
+	"versions":  true,
+}
+
 func main() {
 	args := os.Args[1:]
-	if len(args) < 3 || (args[0] != "data" && args[0] != "shape" && args[0] != "versions") {
-		fmt.Fprintln(os.Stderr, "Usage: iaviewer <data|shape|versions> <leveldb dir> <prefix> [version number]")
-		fmt.Fprintln(os.Stderr, "<prefix> is the prefix of db, and the iavl tree of different modules in cosmos-sdk uses ")
-		fmt.Fprintln(os.Stderr, "different <prefix> to identify, just like \"s/k:gov/\" represents the prefix of gov module")
+	if len(args) < 3 || len(args) > 4 || !cmds[args[0]] {
+		fmt.Fprintln(os.Stderr, strings.TrimSpace(`
+Usage: iaviewer <data|data-full|hash|shape|versions> <leveldb dir> <prefix> [version number]
+<prefix> is the prefix of db, and the iavl tree of different modules in cosmos-sdk uses
+different <prefix> to identify, just like "s/k:gov/" represents the prefix of gov module
+`))
 		os.Exit(1)
 	}
 
-	version := 0
-	if len(args) == 4 {
+	version := int64(0)
```

Comment:
personal preference: a more verbose name would be more readable. 
```suggestion
	wantVersion := int64(0)
```